core/src/main/java/nl/nn/adapterframework/util/ClassUtils.java

```diff
@@ -293,29 +293,70 @@ public static String classNameOf(Object o) {
 		return simpleName;
 	}
 
+	/**
+	 * Throws IllegalArgumentException if the argument type is incompatible
+	 * Throws IllegalStateException if the argument cannot be set on the target bean
+	 */
 	public static void invokeSetter(Object bean, Method method, String valueToSet) {
 		if(!method.getName().startsWith("set") || method.getParameterTypes().length != 1) {
 			throw new IllegalArgumentException("method must start with [set] and may only contain [1] parameter");
 		}
 
 		try {//Only always grab the first value because we explicitly check method.getParameterTypes().length != 1
-			Object castValue = castAsType(method.getParameterTypes()[0], valueToSet);
+			Object castValue = parseValueToSet(method, valueToSet);
 			if(log.isDebugEnabled()) log.debug("trying to set method ["+method.getName()+"] with value ["+valueToSet+"] of type ["+castValue.getClass().getCanonicalName()+"] on ["+ClassUtils.nameOf(bean)+"]");
 
 			method.invoke(bean, castValue);
-		} catch (Exception e) {
-			throw new IllegalArgumentException("error while calling method ["+method.getName()+"] on ["+ClassUtils.nameOf(bean)+"]", e);
+		} catch (IllegalAccessException | InvocationTargetException e) {
+			throw new IllegalStateException("error while calling method ["+method.getName()+"] on ["+ClassUtils.nameOf(bean)+"]", e);
+		}
+	}
+
+	private static Object parseValueToSet(Method m, String value) throws IllegalArgumentException {
+		Class<?> setterArgumentClass = m.getParameters()[0].getType();
+		char[] c = m.getName().substring(3).toCharArray();
+		c[0] = Character.toLowerCase(c[0]);
+		String fieldName = new String(c);
+
+		//Try to parse the value as an Enum
+		if(setterArgumentClass.isEnum()) {
+			return parseAsEnum(setterArgumentClass, fieldName, value);
 		}
+
+		//Try to parse as primitive
+		return convertToType(setterArgumentClass, fieldName, value);
 	}
 
-	private static Object castAsType(Class<?> type, String value) {
+	private static Object convertToType(Class<?> type, String fieldName, String value) throws IllegalArgumentException {
 		String className = type.getName().toLowerCase();
-		if("boolean".equals(className))
-			return Boolean.parseBoolean(value);
-		else if("int".equals(className) || "integer".equals(className))
-			return Integer.parseInt(value);
-		else
-			return value;
+		try {
+			switch (type.getTypeName()) {
+			case "int":
+			case "java.lang.Integer":
+				return Integer.parseInt(value);
+			case "long":
+				return Long.parseLong(value);
+			case "boolean":
+			case "java.lang.Boolean":
+				return Boolean.parseBoolean(value);
+			default:
```

Comment:
`default: return value`, of `"java.lang.String": return value; default: throw new IllegalArgumentExceptiopn(...)`? Om onverwachte return-types meteen af te vangen hier waar het misgaat?